internal/build/container_updater.go

```diff
@@ -70,21 +69,19 @@ func (r *containerUpdater) UpdateInContainer(ctx context.Context, cID containerI
 	return nil
 }
 
-// TODO(maia): reorg tar funcs in a more logical way
 func ArchivePathsIfExist(ctx context.Context, paths []pathMapping) (*bytes.Buffer, error) {
-	buf := new(bytes.Buffer)
-	tw := tar.NewWriter(buf)
+	ab := newArchiveBuilder()
 	defer func() {
-		err := tw.Close()
+		err := ab.close()
 		if err != nil {
 			log.Printf("Error closing tar writer: %s", err.Error())
 		}
```

Comment:
why don't we put the error printing in `ab.close()` so we only need to remember to defer one thing?

Reply:
(tho this is maybe irrelevant if we stop deferring `close()` calls like i'm suggesting in my comment on the PR)